resilience4j-kotlin/src/test/kotlin/io/github/resilience4j/kotlin/circuitbreaker/FlowCircuitBreakerTest.kt

```diff
@@ -0,0 +1,202 @@
+/*
+ *
+ *  Copyright 2019 authors
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *
+ */
+package io.github.resilience4j.kotlin.circuitbreaker
+
+
+import io.github.resilience4j.circuitbreaker.CallNotPermittedException
+import io.github.resilience4j.circuitbreaker.CircuitBreaker
+import kotlinx.coroutines.*
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.toList
+import org.assertj.core.api.Assertions
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Test
+import java.lang.IllegalStateException
+import java.util.concurrent.Phaser
+
+class FlowCircuitBreakerTest {
```

Comment:
Could you please add a test which verifies that the producer is not invoked, when a CircuitBreaker is open.
This test should verify that the `CallNotPermittedException` is thrown, before (in RX terms) a subscriber subscribes to the publisher.

Reply:
This test could help you to decide if `onStart` or `flow` is the correct way to implement the circuitbreaker operator.

Reply:
This case is covered in the `should not collect when open` test but I can see that its not immediately clear. Ill add another test to make it more explicit that the case is covered